src/main/java/duke/Task.java

```diff
@@ -0,0 +1,73 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a task. A Task object corresponds to a task that represented by
+ * its completion status(done),description and deadline
+ */
+
+public class Task {
+    private boolean isDone;
+    private String description;
+    private LocalDate deadline;
+
+    /**
+     * initialises an task object with the given description
+     * 
+     * @param description description of task
+     */
+
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+        this.deadline = null;
+    }
+
+    /**
+     * initialises an task object with the given description and deadline
+     * 
+     * @param description description of task
+     * @param deadline    deadline of task
+     */
+
+    public Task(String description, LocalDate deadline) {
+        this.description = description;
+        this.isDone = false;
+        this.deadline = deadline;
+    }
+
+    public String mark() {
+        this.isDone = true;
+        return this.toString();
+    }
+
+    public String unMark() {
+        this.isDone = false;
+        return this.toString();
+    }
+
+    public Boolean contains(String searchString) {
```

Comment:
Isn't shouldContains better here as 
Boolean variables/methods should be named to sound like booleans?